url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/Events.java

```diff
@@ -0,0 +1,249 @@
+package ccproxy;
+
+import org.apache.log4j.Logger;
+
+import pl.cc.core.PauseType;
+import pl.cc.exceptions.AgentNotFoundException;
+import pl.cc.real.RealAgent;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class Events {
+	org.apache.log4j.Logger log = Logger.getLogger("event");
+	Parametry p;
+
+	public Events(Parametry p) {
+		this.p = p;
+	}
+
+	public void login(AgentConnection ac) {
+		String eventMsg = new String("EVENT [Agent], Action [Login], Agent ["
+				+ ac.agentNumber + "], Exten [" + ac.agentExten + "]"
+				+ ", Name [" + ac.agentName + "]" + ", ID [" + ac.idConnection
+				+ "]");
+		log.info(eventMsg);
+		p.agenci.pushSuperMessage(eventMsg);
+		if (ac.isNewConnected() && p.c.pauseWhenStartup){
+			String defaultPause = p.pauseTypeList.getDefault().getName();
+			String agentPaused = "EVENT [Agent], Action [Pause], " +
+					"Agent ["+ac.agentNumber+"], Exten ["+ac.agentExten+"], Name ["+ac.agentName+"], " +
+					"Type ["+defaultPause+"]";
+			
+			p.agenci.pushSuperMessage(agentPaused);	
+		}
+		
+	}
+
+	public void logout(AgentConnection ac) {
+		String eventMsg = new String("EVENT [Agent], Action [Logout], Agent ["
+				+ ac.agentNumber + "], Exten [" + ac.agentExten + "]"
+				+ ", Name [" + ac.agentName + "]" + ", ID [" + ac.idConnection
+				+ "]");
+		log.info(eventMsg);
+		p.agenci.pushSuperMessage(eventMsg);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 39:44, 48:54, 59:65.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.